code/modules/economy/ATM.dm

```diff
@@ -115,20 +115,20 @@ log transactions
 		return
 	if(!linked_db)
 		reconnect_database()
-	ui_interact(user)
+	tgui_interact(user)
 
 /obj/machinery/atm/attack_ghost(mob/user)
-	ui_interact(user)
+	tgui_interact(user)
 
-/obj/machinery/atm/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+/obj/machinery/atm/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
 	user.set_machine(src)
```

Comment:
```suggestion

```
TGUI removes the requirement for `set_machine()` (and so does nano, this is just super old code)